src/openlcb/SimpleStack.cxx

```diff
@@ -223,24 +229,28 @@ int SimpleStackBase::create_config_file_if_needed(const InternalConfigData &cfg,
     struct stat statbuf;
     bool reset = false;
     bool extend = false;
-    int fd = ::open(CONFIG_FILENAME, O_RDONLY);
+    int fd = configUpdateFlow_.get_fd();
```

Comment:
i think we should rather assert that get_fd() < 0.
create_config_file_if_needed is always the first command in appl_main() when we have an app that operates on a filesystem.

It is not necessary to support calling create_config_file_if_needed after a config file is already open.

Reply:
No problem tidying that up. I'll push it and the `get_fd()` comment through shortly.

I had wondered about whether testing for the file being opened in `SimpleStackBase::create_config_file_if_needed()` made sense as I wrote it. Similar in `SimpleStackBase::start_stack()` but it seems like we can't be completely certain of the order in that case, so it does the test.